upcloud/service/network_test.go

```diff
@@ -817,17 +817,17 @@ func TestCreateNetworkAndServer_DHCPOptions(t *testing.T) {
 
 			assert.ElementsMatch(t,
 				[]string{"172.16.0.0/22"},
-				ipNet.DHCPRoutesConfiguration.EffectiveRoutesAutoPopulation.FilterByDestination,
+				*ipNet.DHCPRoutesConfiguration.EffectiveRoutesAutoPopulation.FilterByDestination,
 			)
 
 			assert.ElementsMatch(t,
 				[]upcloud.NetworkRouteType{"service"},
-				ipNet.DHCPRoutesConfiguration.EffectiveRoutesAutoPopulation.FilterByRouteType,
+				*ipNet.DHCPRoutesConfiguration.EffectiveRoutesAutoPopulation.FilterByRouteType,
 			)
 
 			assert.ElementsMatch(t,
 				[]upcloud.NetworkRouteSource{"static-route"},
-				ipNet.DHCPRoutesConfiguration.EffectiveRoutesAutoPopulation.ExcludeBySource,
+				*ipNet.DHCPRoutesConfiguration.EffectiveRoutesAutoPopulation.ExcludeBySource,
 			)
 		})
```

Comment:
Adding delete to make test runs smoother:

```suggestion
		})
		// Stop the server
		t.Logf("Stopping server with UUID %s ...", srv.UUID)
		err = stopServer(ctx, rec, svc, srv.UUID)
		require.NoError(t, err)
		t.Log("Server is now stopped")

		// Delete the server and storage
		t.Logf("Deleting the server with UUID %s, including storages...", srv.UUID)
		err = deleteServerAndStorages(ctx, svc, srv.UUID)
		require.NoError(t, err)
		t.Log("Server is now deleted")

		// Delete the network
		t.Logf("Deleting the network with UUID %s...", network.UUID)
		err = svc.DeleteNetwork(ctx, &request.DeleteNetworkRequest{
			UUID: network.UUID,
		})
		require.NoError(t, err)
		t.Log("Network is now deleted")
```